### PR TITLE
Make lots of services private

### PR DESCRIPTION
**Pull request by WouterJ**
All these services aren't created for public usage. By making them private, they'll be removed during compilation time once no other service depends on it and if only one service depends on it, initialization will be inlined. Both improve the container performance.

Besides this, it has also become a best practice.

### CODE REVIEW

File: Resources/config/publish-workflow.xml [hunk 1]
```diff
@@ -29,34 +29,34 @@
...
             <tag name="cmf_published_voter" priority="10"/>
         </service>
 
-        <service id="cmf_core.security.publish_time_period_voter" class="%cmf_core.security.publish_time_period_voter.class%">
+        <service id="cmf_core.security.publish_time_period_voter" class="%cmf_core.security.publish_time_period_voter.class%" public="false">
             <tag name="cmf_published_voter" priority="20"/>
         </service>
 
-        <service id="cmf_core.publish_workflow.request_listener" class="%cmf_core.publish_workflow.request_listener.class%">
+        <service id="cmf_core.publish_workflow.request_listener" class="%cmf_core.publish_workflow.request_listener.class%" public="false">
```
Comment (dbu) [inline Resources/config/publish-workflow.xml]: i think this leads to 

[InvalidArgumentException]

The service "cmf_core.publish_workflow.request_listener" must be public as event subscribers are lazy-loaded.

**Comment by dbu** [comment]
i agree that we should do this. there is one problem though, its a BC break if people relied on those services to be available from the container previously. we need to bump the version. imho, minor version is enough as its only breaking when people did things the wrong way... we should add a changelog saying which services accidentally where public until now so people can check their codebase.

File: Resources/config/form-type.xml [hunk 2]
```diff
@@ -8,7 +8,7 @@
     </parameters>
 
     <services>
-        <service id="cmf_core.form.type.checkbox_url_label" class="%cmf_core.form.type.checkbox_url_label.class%">
+        <service id="cmf_core.form.type.checkbox_url_label" class="%cmf_core.form.type.checkbox_url_label.class%" public="false">
```
Comment (dbu) [inline Resources/config/form-type.xml]: is it not a legal use case to use a form type by service name?

File: Resources/config/services.xml [hunk 3]
```diff
@@ -12,15 +12,16 @@
...
             <argument type="service" id="cmf_core.templating.helper"/>
             <tag name="twig.extension"/>
         </service>
 
-        <service id="cmf_core.templating.helper" class="%cmf_core.templating.helper.class%">
+        <service id="cmf_core.templating.helper" class="%cmf_core.templating.helper.class%" public="false">
             <argument type="service" id="cmf_core.publish_workflow.checker" on-invalid="ignore"/>
             <argument type="service" id="doctrine_phpcr" on-invalid="ignore"/>
             <argument>%cmf_core.persistence.phpcr.manager_name%</argument>
+            <tag name="templating.helper" alias="cmf"/>
```
Comment (dbu) [inline Resources/config/services.xml]: does this mean that until now, the helper was not registered with templating and people either used the service name or nobody used the templating helper?

Comment (WouterJ) [inline Resources/config/services.xml]: this means nobody used the templating helper

File: Resources/config/form-type.xml (hunk 2 above)
Comment (WouterJ) [inline Resources/config/form-type.xml]: in such case, you refer to it by its name. It is already loaded into the service in that case.

**Comment by WouterJ** [comment]
@dbu it would still work if services relied on it. The only thing that doesn't work is when you use `$container->get()` to get one of these services. I don't think anyone did that. Anyway, this PR already targets 1.3

**Comment by dbu** [comment]
CheckboxUrlLabelFormTypeTest seems to have ignored the hint about using the form type name... can you adjust that? and can you please add all service names that became private to the changelog to simplify debugging when updating?

**Comment by lsmith77** [comment]
@dbu: master branch is for the next minor release

**Comment by dbu** [comment]
yep, saw that. i think its ok to do this change if we have it in the changelog. its only cleaning up something, but with a risk of errors if people where using services they where not supposed to use. @lsmith77 do you agree with that or think we can only change this in a 2.0?

**Comment by lsmith77** [comment]
I think its ok to do this in a minor release

**Comment by dbu** [comment]
@WouterJ can you add the changelog and investigate the travis error? 1) Symfony\Cmf\Bundle\CoreBundle\Tests\Functional\Form\CheckboxUrlLabelFormTypeTest::testFormTwigTemplate
Symfony\Component\DependencyInjection\Exception\ServiceNotFoundException: You have requested a non-existent service "cmf_core.form.type.checkbox_url_label".

i guess we need to do change that test...

**Comment by WouterJ** [comment]
Updated CHANGELOG and fixed the tests (form type services can't be private either).

**Comment by dbu** [comment]
we get a strange test failure regarding the "--force" parameter. can you check what this is?

**Comment by WouterJ** [comment]
@dbu something weird is happening: By default, it installs doctrine/dbal 2.4 (causing a version jackalope to be installed which doesn't have `--force`). However, if you add `doctrine/dbal: 2.5.*`, it just installs 2.5 for you, removing all problems :)

I've seen this happening in the ResourceBundle too (it's failing there as well).

We should update the Testing component to also support older jackalope version btw, but I haven't done that yet as the error is a clear indicator that a wrong doctrine/dbal version is installed.

**Pull request merged**